src/wayland/mod.rs

```diff
@@ -15,7 +15,7 @@ struct State;
 
 pub struct WaylandDisplayImpl {
     conn: Connection,
-    event_queue: Mutex<EventQueue<State>>,
+    event_queue: RefCell<EventQueue<State>>,
```

Comment:
You don't need this `RefCell`, so remove it.

Reply:
As things are currently implemented the `EventQueue` is part of the `Rc<WaylandDisplayImpl>` shared by all surfaces using the display. And it needs a mutable references to dispatch. So `RefCell` is needed.

Reply:
Aye.